functions/src/index.ts

```diff
@@ -128,4 +143,52 @@ app.post('/registerWithEmail', (req: Request, res: Response) => {
     })
 })
 
+/**
+ * API for retrieving user data
+ *
+ * @req JSON containing "uid" field which is the unique identifier for the user
+ * @res Stores the status of the request and a json containing either
+ *      the "error" field if the API crashed, the "general" field, or the user data
+ *      The following may be stored in the general field:
+ *          - User Data not Found: Usually means that this user was created before
+ *                                 the new database was created for user data
+ *      User data fields may include:
+ *          - email
+ *          - username
+ *          - name
+ *          - problemsAttempted
+ *          - problemsAccepted
+ *          - problemsRTE
+ *          - problemsTLE
+ *          - ProblemsWrong
+ */
+app.post('/getUserData', (req: Request, res: Response) => {
+  const userId: string = req.body.uid
+  admin
+    .auth()
+    .getUser(userId)
+    .then((curUser) => {
+      let data = {
+        email: curUser.email,
+        username: curUser.displayName,
+      }
+
+      getDoc(doc(db, 'UserData', userId))
+        .then((doc) => {
+          if (doc.exists()) {
+            data = { ...data, ...doc.data() }
```

Comment:
...Merging the two JSONs from two different sources to send back a single package. I see why this was a tricky thing to do.